docs/DeveloperGuide.md

```diff
@@ -1187,28 +1196,92 @@ The `MathUtil` class contains generic functions for mathematical operations such
 
 Each individual unit of the program is tested to ensure each piece works correctly and stubs are commonly used to test the units in isolation.
 
-### **Testing Models**
+Unit tests were implemented as new features were added in, to ensure that any new changes will have tests to ensure 
+older features are working as intended. Unit tests were also changed and added on as the product progressed, to ensure
+that the features were tested more thoroughly as the team learned new testing types and techniques.
+
+
+### **Stubs**
+
+Stubs were used to test features and classes in isolation, so that any bugs in dependencies will not affect the tests
+that are in place. This is most common in the testing of commands as well as parsers, to ensure that any bugs in one or
+the other will not result in failing tests in the other, allowing us to quickly ascertain bugs without having to find 
+out where the bug originated from. 
+
+We replaced the parser with stubs to test the commands and vice versa.
+
+### **Integration Testing**
 
-Models used:
-* Waterfall model
-* Agile model
-* Breadth-first iterative model
+Integration testing was used extensively in the testing of the GUI, as part of CI. Integration testing were also 
+included in unit tests to ensure that our features work well with each other. It was also covered manually via developer
+testing and dogfooding.
 
-We used a mixture of models to suit our needs for different features. We applied the waterfall model to features that required individual work and the agile model for the others that required teamwork. We also had considerations for the time-limit imposed on this project so we applied the breadth-first iterative model to ensure basic functionalities.
 
-### **Testing Commands**
+### **Dogfooding**
 
-Equivalence partitions and stubs were mainly used to test commands.
+Towards the completion of our product, our developers used the product on a regular basis in an attempt to use the 
+product how a normal user might use. This helped ensure that our product would be relatively bug-free for a regular use
+case.
 
-### **Testing Parsers**
+## **Model**
 
-Positive and negative test cases were mainly used to test parsers.
+Our organisation created dummy values for a model of a simulated list of persons that our product might, and used that
+to carry out testing on. When changes were made to the model in the product, any changes required were reflected in 
+this testing model as well, to ensure that unit tests are testing on a correct simulation of the program.
+
+## **Testing Commands**
+
+Our organisation tested commands by creating commands with the correct parameters, bypassing the parser, and executed 
+them. Assertions were used to ensure that commands created, when executed performed the correct operations, on the model
+created specifically for tests. 
```

Comment:
It is also not just correct I/O, it also ensures proper exceptions were thrown as and when needed.

Reply:
edited.